src/pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -14,7 +14,7 @@
 import multiprocessing
 import os
 from dataclasses import dataclass, field
-from typing import Any, Callable, Collection, List, Optional, Tuple, Union
+from typing import Any, Callable, Collection, Iterable, List, Optional, Tuple, TypeAlias, Union
```

Comment:
Hi @JustinGoheen, all of the tests are now failing, as none of them are running on 3.10, which is the version when `TypeAlias` became available. https://docs.python.org/3/library/typing.html#typing.TypeAlias In order to fix it, please import `TypeAlias` from `typing_extensions`

Reply:
updated in https://github.com/Lightning-AI/lightning/pull/13806/commits/87cb7a66d3ed03cdca5ed8ae60d211a7049e329e

Reply:
Well, another error which made every test broke popped up -> `TypeAlias` is not subscriptable

Reply:
TypeAlias has been removed